pkg/util/pool/bucketed_pool_test.go

```diff
@@ -16,48 +16,109 @@ func makeFunc(size int) []int {
 }
 
 func TestBucketedPool_HappyPath(t *testing.T) {
-	testPool := NewBucketedPool(1, 8, 2, makeFunc)
+
 	cases := []struct {
 		size        int
 		expectedCap int
 	}{
 		{
-			size:        -1,
+			size:        0,
+			expectedCap: 0,
+		},
+		{
+			size:        1,
 			expectedCap: 1,
 		},
 		{
+			// One less than bucket boundary
 			size:        3,
 			expectedCap: 4,
 		},
 		{
-			size:        10,
-			expectedCap: 10,
+			// Same as bucket boundary
+			size:        4,
+			expectedCap: 4,
+		},
+		{
+			// One more than bucket boundary
+			size:        5,
+			expectedCap: 8,
+		},
+		{
+			// Two more than bucket boundary
+			size:        6,
+			expectedCap: 8,
+		},
+		{
+			size:        8,
+			expectedCap: 8,
+		},
+		{
+			size:        16,
+			expectedCap: 16,
+		},
+		{
+			size:        20,
```

Comment:
(nit) worth a comment explaining why this isn't expecting 36

Reply:
Added in 0552399.